### PR TITLE
Display collage count before taking photo

### PR DESCRIPTION
**Pull request by rawbertp**
**Current behaviour**:

- `L10N.cheeseCollage` is shown ('Cheeeeeeeese Collageeeeeeeeeee!')
- Picture is taken
- '1/4' is displayed
- `L10N.cheeseCollage` is shown
- Picture is taken
- '2/4' is displayed
- `L10N.cheeseCollage` is shown
- Picture is taken
- '3/4' is displayed
- `L10N.cheeseCollage` is shown
- Picture is taken
- Picture gets processed
- Where is "4/4"?

**Improvement**:

I find this quite counterintuitive and suggest the following change which displays the photo # **before** the picture is taken, so it's like: '1/4' - Pic - '2/4' - Pic ... '4/4' - Pic

- `L10N.cheeseCollage` + "1/4" is shown
- Picture is taken
- `L10N.cheeseCollage` + "2/4" is shown
- Picture is taken
- `L10N.cheeseCollage` + "3/4" is shown
- Picture is taken
- `L10N.cheeseCollage` + "**4/4**" is shown
- Picture is taken
- Picture gets processed

### CODE REVIEW

File: resources/js/core.js [hunk 1]
```diff
@@ -154,6 +155,7 @@ const photoBooth = (function () {
             $('.loading').text(L10N.cheese);
         } else {
             $('.loading').text(L10N.cheeseCollage);
+            $('<p>').text(`${nextCollageNumber + 1} / ${collageLimit}`).appendTo('.loading');
```
Comment (andi34) [inline resources/js/core.js]: `${result.limit}` instead ?

Comment (rawbertp) [reply resources/js/core.js]: Nope, that won't work unfortunately as `result` is no global variable and hence not available within that scope. Mid term both this newly introduced `collageLimit` as well as `result.limit` (which is in turn hard-coded `4` in `takePic.php` should be put into `lib/config.php` (I guess).

Comment (andi34) [reply resources/js/core.js]: Mh... Ok
Putting it in config.php would mean we need to allow different collage options (e.g. collage out of 2 pictures, out of 3 pictures etc.). This means the whole collage function needs to be reworked to always generate a collage.

Comment (rawbertp) [reply resources/js/core.js]: I probably put it wrong... :) I wasn't about to suggest it should be configurable but it should be available as a global/static/constant variable.

**Comment by andi34** [comment]
@rawbertp i've pushed the eslint fix here https://github.com/andreknieriem/photobooth/pull/193

**Comment by rawbertp** [comment]
> @rawbertp i've pushed the eslint fix here #193

Ok thanks, I have rebased the branch.

File: resources/js/core.js [hunk 2]
```diff
@@ -15,7 +15,8 @@ const photoBooth = (function () {
                 height: 480,
                 facingMode: 'user',
             }
-        };
+        },
+        collageLimit = 4;
```
Comment (sualko) [inline resources/js/core.js]: Now we have to limits. One in the php script and one here. Such thing should be maintained at one place.

File: resources/js/core.js (hunk 1 above)
Comment (sualko) [inline resources/js/core.js]: This adds the picture number to the cheese message and is distinct from what you described in the pr description (at least in my understanding).

File: resources/js/core.js [hunk 3]
```diff
@@ -192,12 +194,11 @@ const photoBooth = (function () {
 
                 $('.spinner').hide();
                 $('.loading').empty();
-                $('<p>').text(`${result.current + 1} / ${result.limit}`).appendTo('.loading');
 
                 if (config.continuous_collage) {
                     setTimeout(() => {
                         public.thrill('collage');
-                    }, 1000);
+                    }, 0);
```
Comment (sualko) [inline resources/js/core.js]: This is completely independent from your pr and the better solution would be to make this configurable.

**Comment by rawbertp** [comment]
> For me your implementation differs from your description and I don't get it which benefit I get.

For me the information that the picture taken was "1/4", "2/4", "3/4" doesn't make sense. I find it more useful to have the information upfront. Also, I'm missing the "4/4" (I might be picky but for me this is not "complete", something is missing). But that might be just me and if you do not find it useful that way, I will just implement it in my fork - it's nothing crucial, rather a matter of taste.

File: resources/js/core.js (hunk 1 above)
Comment (rawbertp) [reply resources/js/core.js]: I updated the description - maybe it makes more sense now?

**Comment by sualko** [comment]
> which displays the photo # before the picture is taken

I read over the number symbol. Sorry.

I see the point that 4/4 is not displayed and that this could enhance the ux. To merge this you should revert the changed timeout or make it configurable and add the number of photos in a collage as hard coded config to `lib/config.php` so that it's accessible in js and php.

**Comment by rawbertp** [comment]
No worries, will do!

File: resources/js/core.js (hunk 2 above)
Comment (rawbertp) [reply resources/js/core.js]: Done - can you please review the changes again?

File: resources/js/core.js (hunk 3 above)
Comment (rawbertp) [reply resources/js/core.js]: I reverted this.

**Comment by andi34** [comment]
@rawbertp could you please update the commit message?
`Fix an eslint error in pho……toinit.js` isn't needed

**Comment by andi34** [comment]
Somehow this does not work for me. it's telling `1/false` and failing.
To workaround that https://github.com/andi34/photobooth/commit/68321b1614462579013142594cd8ccac55a35ac4

**Comment by sualko** [comment]
Looks good. Had no time to test it.

@andi34 I think it's a bad idea to add an option you are not allowed to change. Can you try to debug why you get a false als collage limit?

**Comment by andi34** [comment]
Tried different things but couldn't find the reason... Maybe because it's not in config setup PHP?

**Comment by rawbertp** [comment]
Doesn't make sense to me. I even tested this with Apache http now as this would be the only difference I can think of but that also works for me.

`docker run -dit --name my-apache-php -p 8080:80 -v "$PWD":/var/www/html php:7.2-apache`

**Comment by andi34** [comment]
I've tested in dev mode on Xubuntu using Apache

**Comment by andi34** [comment]
@rawbertp you've changed and saved config via admin panel? I think the issue starts once my.config.inc.php gets generated.

**Comment by andi34** [comment]
Ok, I've an idea. 
Guess this should be moved on line 83 or below (faster deep merge happened already).
Will test and report back

**Comment by rawbertp** [comment]
@andi34 ok you're right, I haven't tested that unfortunately.
It appears to be caused by this LOC: https://github.com/andreknieriem/photobooth/blob/master/api/admin.php#L66

I might be just too tired right now but I don't really get its purpose. Commenting it out fixes the issue but probably causes another one?! @sualko maybe you can advise?

**Comment by rawbertp** [comment]
I committed a fix but not sure if you are going to like it... 🙈

**Comment by sualko** [comment]
Why do we need a hidden input?

**Comment by andi34** [comment]
@sualko See my comments above. On save it will default to "false" without

**Comment by rawbertp** [comment]
What @andi34 says. And I'm not saying that this is the best solution, it just feels better/more correct to me than applying any "workarounds". As far as I understand it the problem is the differentiation between items that are not in the form at all (such as the new ``collage_limit`` and checkboxes that are not ticked and thus not part of the posted form-data). The current implementation checks for settings declared and if not found in the form data they will be considered as 'false'.

**Comment by andi34** [comment]
i'd prefer the hidden config over my fix to avoid people changing the config

**Comment by andi34** [comment]
@sualko any better idea? 

we've 2 solutions, guess the one from @rawbertp is the better choice
1. https://github.com/andi34/photobooth/commit/68321b1614462579013142594cd8ccac55a35ac4
2. (prefered) https://github.com/andreknieriem/photobooth/pull/200/commits/f417fa94525a28ea8095e8a3e00d101b76d1ee26

Source of need for one of both solutions to be needed:
> @andi34 ok you're right, I haven't tested that unfortunately.
> It appears to be caused by this LOC: https://github.com/andreknieriem/photobooth/blob/master/api/admin.php#L66
> 
> I might be just too tired right now but I don't really get its purpose. Commenting it out fixes the issue but probably causes another one?! @sualko maybe you can advise?
> 
> Update: Figured it out myself. If a checkbox is not ticket it's not included in the form data.

**Pull request merged**